cmd/convert.go

```diff
@@ -18,61 +18,73 @@ var (
 	convertCmdAssumeYes         bool
 )
 
+func executeConvert(_ *cobra.Command, _ []string) error {
+	sourceFormat, err := convert.ParseFormat(convertCmdSourceFormat)
+	if err != nil {
+		return err
+	}
+	destinationFormat, err := convert.ParseFormat(convertCmdDestinationFormat)
+	if err != nil {
+		return err
+	}
+
+	if convertCmdInputFile != "" {
+		if yes, err := utils.ConfirmFileOverwrite(
+			convertCmdOutputFile, "", convertCmdAssumeYes,
+		); err != nil {
+			return err
+		} else if !yes {
+			return nil
+		}
+
+		err = convert.Convert(convertCmdInputFile, convertCmdOutputFile, sourceFormat, destinationFormat)
+		if err != nil {
+			return fmt.Errorf("converting file: %w", err)
+		}
+	} else if is2xTo3xConversion() {
+		path, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("getting current working directory: %w", err)
+		}
+		files, err := utils.ConfigFilesInDir(path)
+		if err != nil {
+			return fmt.Errorf("getting files from directory: %w", err)
+		}
+		for _, filename := range files {
+			err = convert.Convert(filename, filename, sourceFormat, destinationFormat)
+			if err != nil {
+				return fmt.Errorf("converting '%s' file: %w", filename, err)
+			}
+		}
+	}
+	if convertCmdDestinationFormat == "konnect" {
+		cprint.UpdatePrintf("Warning: konnect format type was deprecated in v1.12 and it will be removed\n" +
+			"in a future version. Please use your Kong configuration files with deck <cmd>.\n" +
+			"Please see https://docs.konghq.com/konnect/getting-started/import/.\n")
+	}
+	return nil
+}
+
 // newConvertCmd represents the convert command
-func newConvertCmd() *cobra.Command {
+func newConvertCmd(deprecated bool) *cobra.Command {
+	short := "Convert files from one format into another format"
+	execute := executeConvert
+	if deprecated {
+		short = "[deprecated] use 'file convert' instead"
+		execute = func(cmd *cobra.Command, args []string) error {
+			cprint.UpdatePrintf("Warning: the 'deck convert' command was deprecated and moved to 'deck file convert'.\n")
```

Comment:
Nit: "deprecated" is already past tense. "was deprecated" should be "is deprecated" which indicates a field is currently marked deprecated.  "was" is less clear

Reply:
I also think we should indicate that the alias still works but will be removed in a future version. 

```
'deck convert' is DEPRECATED and will be removed in a future version. Use 'deck file convert' instead.
```